apps/aecore/src/aec_peers_pool.erl

```diff
@@ -0,0 +1,1695 @@
+%%%=============================================================================
+%%% @copyright 2017, Aeternity Anstalt
+%%% @doc Stochastic pool.
+%%%
+%%% Opaque data structure that handle a collection of peers.
+%%% Implements the design described in:
+%%%   https://github.com/aeternity/protocol/blob/master/GOSSIP.md
+%%%
+%%% Usage:
+%%%  - When setting up, add trusted peers calling `update/7`.
+%%%  - When receiving a peer through gossip, call `update/7`.
+%%%  - To select a peer to connect to, call `select/4`.
+%%%  - When an outbound connection failed, call `reject/3`.
+%%%  - When an outbound connection is closed, call `release/3`.
+%%%  - When an inbound connection is established and the first gossip
+%%%    message is received, call `update/7`, `verify/3` and `selected/4`.
```

Comment:
I feel this bullet point could be explained in more detail. When should the individual calls be made, and what are the error scenarios (i.e. should `reject/3` be called somewhere in-between in some cases)?